### PR TITLE
Added Live Content shortcode

### PR DESCRIPTION
**Pull request by rafsuntaskin**
JIRA Ticket: https://moderntribe.atlassian.net/browse/EXT-156
**[tribe-event-live-content]**  shortcode added to show content between the start and end times of the event only.

Example usage:
http://p.tri.be/cPeh3g

### CODE REVIEW

File: src/Tribe/Live_Content.php [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+	 *
+	 * @return string
+	 */
+	public function render_content( $event ) {
+
+		$times = $this->get_times( $event );
+
+		// if no valid time is set.
+		if ( empty( $times ) ) {
+			return __( 'No Start Time is set for Live Content to appear.', 'tribe-ext-events-happening-now' );
```
Comment (cliffordp) [inline src/Tribe/Live_Content.php:115]: consider changing to `_x(...)`

File: readme.txt [hunk 2]
```diff
@@ -43,6 +43,10 @@ We're always interested in your feedback and our [Help Desk](https://support.the
 
 == Changelog ==
 
+= [TBD] TBD =
+
+* TBD
+
```
Comment (cliffordp) [inline readme.txt:49]: Add your changelog entry

File: readme.txt [hunk 3]
```diff
@@ -5,7 +5,7 @@ Tags: events, calendar
 Requires at least: 4.5
 Tested up to: 5.3.2
 Requires PHP: 5.6
-Stable tag: 1.0.0
+Stable tag: TBD
```
Comment (cliffordp) [inline readme.txt:8]: add your version

File: src/views/happening-now.php [hunk 4]
```diff
@@ -10,7 +10,7 @@
  *
  * @link {INSERT_ARTICLE_LINK_HERE}
  *
- * @version 5.0.2
+ * @version TBD
```
Comment (cliffordp) [inline src/views/happening-now.php:13]: when this template was added

File: tribe-ext-events-happening-now.php [hunk 5]
```diff
@@ -4,7 +4,7 @@
  * Plugin URI:        https://theeventscalendar.com/extensions/events-happening-now/
  * GitHub Plugin URI: https://github.com/mt-support/tribe-ext-events-happening-now
  * Description:       Provides a widget, shortcode, and optional meta for displaying events that are happening now.
- * Version:           1.0.0
+ * Version:           TBD
```
Comment (cliffordp) [inline tribe-ext-events-happening-now.php:7]: your version number

File: readme.txt [hunk 6]
```diff
@@ -5,7 +5,7 @@ Tags: events, calendar
 Requires at least: 4.5
 Tested up to: 5.3.2
```
Comment (cliffordp) [inline readme.txt:6]: update WP version

File: readme.txt [hunk 7]
```diff
@@ -19,7 +19,7 @@ This extension adds the `[tribe-happening-now]` shortcode. The parameters that c
 * `category`: Categories to display
 * `featured`: If true, only includes featured events. If false, excludes featured events. If null (left off the shortcode), includes all.
 * `hide_url`: If true, suppresses the Website URL CTA
-* `quantity`: Max number of events to show (defaults to unlimited “-1”)
+* `quantity`: Max number of events to show (defaults to unlimited "-1")
 * `title`: Title above event list
 * `url_title`: Title of event’s Website URL CTA
```
Comment (cliffordp) [inline readme.txt:25]: Document the new `[tribe-event-live-content]`

File: src/Tribe/Live_Content.php [hunk 8]
```diff
@@ -0,0 +1,422 @@
...
+	 *
+	 * @since TBD
+	 *
+	 * @var   array
+	 */
+	protected $validate_arguments_map = [
+		'id'               => [ self::class, 'validate_event' ],
+		'content_extended' => [ self::class, 'validate_time_string' ],
+		'start_time'       => [ self::class, 'validate_time_string' ],
+		'end_time'         => [ self::class, 'validate_time_string' ],
```
Comment (cliffordp) [inline src/Tribe/Live_Content.php:51]: are all 3 of these the same validation, especially `content_extended`?

File: src/Tribe/Live_Content.php [hunk 9]
```diff
@@ -0,0 +1,422 @@
...
+
+	/**
+	 * Array of callbacks for arguments validation
+	 *
+	 * @since TBD
+	 *
+	 * @var   array
+	 */
+	protected $validate_arguments_map = [
+		'id'               => [ self::class, 'validate_event' ],
```
Comment (cliffordp) [inline src/Tribe/Live_Content.php:48]: Consider renaming from "event" to "post"

File: src/Tribe/Live_Content.php [hunk 10]
```diff
@@ -0,0 +1,422 @@
...
+	 *
+	 * @since TBD
+	 *
+	 * @param $id
+	 *
+	 * @return false|int
+	 */
+	public static function validate_event( $id ) {
+		return ( null === $id || 'null' === $id ) ? get_the_ID() : $id;
+	}
```
Comment (cliffordp) [inline src/Tribe/Live_Content.php:410]: Consider using `tribe_post_exists()`

If null, get current post. If not null, check if post exists - if it doesn't (bad ID passed), have the shortcode return an empty string.

**Review by cliffordp — CHANGES_REQUESTED** [review]
Besides what's in my line-specific comments, you could convert these shortcodes to use Common's recent updates: https://github.com/moderntribe/tribe-common/blob/master/src/Tribe/Shortcode/Shortcode_Abstract.php

Example how-to: https://github.com/moderntribe/event-tickets-plus/tree/release/G20.05/src/Tribe/Shortcode

**Comment by rafsuntaskin** [comment]
I assume this is deprecated now. So, closing the PR to minimize noise for PRs.

**Pull request closed without merging**